src/Pronamic.php

```diff
@@ -80,80 +80,41 @@ public static function get_submission_value( $type ) {
 				continue;
 			}
 
-			$value = array_key_exists( $tag->name, $_POST ) ? \sanitize_text_field( \wp_unslash( $_POST[ $tag->name ] ) ) : '';
+			// Submission value.
+			$value = $submission->get_posted_string( $tag->name );
 
-			$value = trim( $value );
-
-			if ( 'checkbox' === $tag->basetype ) {
-				$value = \filter_input( \INPUT_POST, $tag->name, \FILTER_DEFAULT, \FILTER_REQUIRE_ARRAY );
-			}
-
-			// Check empty value.
 			if ( empty( $value ) ) {
 				continue;
 			}
 
-			$values = (array) $value;
-
-			// Loop values.
-			foreach ( $values as $value ) {
-				/*
-				 * Try to get value from piped field values.
-				 *
-				 * @link https://contactform7.com/selectable-recipient-with-pipes/
-				 */
-				if ( $tag->pipes instanceof WPCF7_Pipes ) {
-					// Make multidimensional array with pipe options by value.
-					$options = [];
-
-					$labels = $tag->pipes->collect_befores();
-
-					foreach ( $tag->pipes->collect_afters() as $key => $after ) {
-						// Make sure array for value exists.
-						if ( ! \array_key_exists( $after, $options ) ) {
-							$options[ $after ] = [];
+			switch ( $type ) {
+				case 'amount':
+					/**
+					 * Contact Form 7 concatenates the field option value with user input for free text fields. We
+					 * are only interested in the input value as amount.
+					 *
+					 * @link https://github.com/rocklobster-in/contact-form-7/blob/2cfaa472fa485c6d3366fcdd80701fdaf7f9e425/includes/submission.php#L434-L437
+					 */
+					if ( \wpcf7_form_tag_supports( $tag->type, 'selectable-values' ) && $tag->has_option( 'free_text' ) ) {
+						$values = \WPCF7_USE_PIPE ? $tag->pipes->collect_afters() : $tag->values;
+
+						$last_value = end( $values );
+
+						if ( \str_starts_with( $value, $last_value . ' ' ) ) {
+							$value = substr_replace( $value, '', 0, strlen( $last_value . ' ' ) );
 						}
-
-						// Add option to value array.
-						$options[ $after ][] = $labels[ $key ];
 					}
 
-					// Search for value in options.
-					foreach ( $options as $after => $labels ) {
-						if ( false !== \array_search( $value, $labels, true ) ) {
-							// Handle free text input.
-							if ( $tag->has_option( 'free_text' ) && end( $tag->values ) === $value ) {
-								$free_text_name = sprintf( '%s_free_text', $tag->name );
-
-								if ( \array_key_exists( $free_text_name, $_POST ) ) {
-									$value = trim( \sanitize_text_field( \wp_unslash( $_POST[ $free_text_name ] ) ) );
-								}
-
-								break;
-							}
-
-							// Set 'after value' as value.
-							$value = $after;
+					$value = Tags\AmountTag::parse_value( $value );
 
-							break;
-						}
+					// Set parsed value as result or add to existing money result.
+					if ( null !== $value ) {
+						$result = ( null === $result ? $value : $result->add( $value ) );
```

Comment:
At what point is `$result` a object with `add` function? PHPStan doesn't like this:

> Cannot call method add() on Pronamic\WordPress\Money\Money|string.

Reply:
When using multiple amounts on the form, for example with the `[pronamic_pay_amount]` tag.

Resolved in https://github.com/pronamic/wp-pronamic-pay-contact-form-7/pull/15/commits/aa98a065b1a7c8282ee877bafbd7d4c7b87613a0?